### ISSUE TITLE
Uncaught exception of type NSException

### ISSUE THREAD

**Issue by Tybi** [wkh237/react-native-fetch-blob#119]
Hi ! I have a problem when I use the sample for uploading to DropBox. In android everything works. 
But in iOS when I upload the app crash in Xcode. The problem appears when the code try to detect if the response header for Content-Type contains "text/" or "application/json". 

This is what I have in XCode log : 

2016-09-04 17:11:09.278 Olly[390:5537] -[__NSCFString containsString:]: unrecognized selector sent to instance 0x14ff5850
2016-09-04 17:11:09.281 Olly[390:5537] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString containsString:]: unrecognized selector sent to instance 0x14ff5850'
*** First throw call stack:
(0x2eb98f83 0x39349ccf 0x2eb9c917 0x2eb9b203 0x2eaea768 0x173685 0x2e81356d 0x2f4dd3c7 0x2f4cd5ab 0x2f57176d 0x398339ab 0x39844921 0x39844b21 0x39973bd3 0x39973a98)
libc++abi.dylib: terminating with uncaught exception of type NSException

The strange things is that the value of respCType is not null and contain "application/json" as you can see in the screenshot : 

<img width="1132" alt="capture d ecran 2016-09-04 a 17 21 08" src="https://cloud.githubusercontent.com/assets/7558089/18231927/d17b8860-72c4-11e6-9684-3e2e996fb5fd.png">

If I check the status code, the value is 200 that mean the upload is done successfully.

I don't understand what is the problem, I'm using rn-0.32. 

Thank you.

**Comment by wkh237** [comment]
@Tybi , thank you for reporting this issue, however, I could not reproduce this issue after some attemps, prehaps I need your assist. 

Could you provide the code snippet you're using ? how's the size of the request body ? what's the version of the library ? Thank you 😄

**Comment by Tybi** [comment]
The version of the library is "react-native-fetch-blob": "^0.9.3" and "react-native": "^0.32.0", 
This is the content of my function : 


```
var encoded = base64.encode(JSON.stringify(backupData));
var response = await RNFetchBlob.fetch('POST', 'https://content.dropboxapi.com/2/files/upload', {
      Authorization : "Bearer "+dbToken,
      'Dropbox-API-Arg': JSON.stringify({
        path : '/data_backup.json',
        mode : 'overwrite',
      }),
      'Content-Type' : 'application/octet-stream',
    }, encoded);
if (response.status !== 200){
      return false;
}
return true;
```

backupData value  is 

`{data:"25ecce720d11cc6a163ff5ee039e8c308c9554728cb996faefd6ace6b4a23ffcU2FsdGVkX19VatGktnLQ3Zsi4f9Qv69F5Un0En7phNEdRYiKOV+qa8om4rxzpouJV5m4mKKBxPt674DrJV64v6RcWfawZGu3qxb7LkRPECYUQMJsuHdQEQ/PsXx4UKO3UZKWUOFaamddjZ9nwYYKVIiAAPiyKYJg7t+qf2m1ZY4="
verification:"d1968bb5c6049277ae45d8fb62c1519a428e8825a639c785238548dd82e6d92cU2FsdGVkX19I5N9ixyTJzHWYUjjGeNW1SKet0OHk1CUJ6WUuVCwlb+Q7XTRC1kEZjJuR8+czCFal4m7XGliY7A=="}`

And the contentLength of the request is the sendRequest method is 417.

**Comment by wkh237** [comment]
Thanks for the detailed information, I'll look into it 👍

**Comment by wkh237** [comment]
@Tybi , I've tested with the code and data you provided, but still can not reproduce the error. The XCode error message looks like it can not find a method named `containsString`, which is a method introduced in IOS 8.0. I only run test on 8.0 devices and simulator, not sure if this is the problem ?

**Comment by Tybi** [comment]
Ok, I will try to change my deployement target because it's 7.0 maybe this is the problem! 

Thanks.

**Comment by Tybi** [comment]
@wkh237 , After some test the problem only appear on my device wich is in IOS 7.0  on the simulator the app doesn't crash. To solve my problem I replace the original code 

```Objective-C
if([respCType containsString:@"text/"]){
     respType = @"text";
}
```
By
```Objective-C
NSRange range = [respCType rangeOfString:@"text/"];

if(range.location != NSNotFound){ 
     respType = @"text";
 }
```

Thanks  👍

**Comment by wkh237** [comment]
Great ! I think we can fix this by extend the NSString class (according to the SO thread) so that it can compatible with IOS 7 😏

**Comment by wkh237** [comment]
Besides, I've noticed that in React Native official repo it says

> Supported operating systems are >= Android 4.1 (API 16) and >= iOS 8.0. 

Not sure if I'm reading this correctly ?

**Comment by Tybi** [comment]
Yes you have right. But it's strange in some issues of React-Native people says it support IOS 7 and before this problem I never encountred problems with IOS 7. 

The thing is I can't find data about the part of Iphone still using IOS 7 and if it's still interesting to support this version.

**Comment by wkh237** [comment]
@Tybi , I've added extended `containsString` to IOS module but I can't verify if it works on IOS7. Could  try install `0.9.4-beta.7` see if it works ? thank you 😄

**Comment by wkh237** [comment]
The fix has been released with `0.9.4` so I'm going to close this issue. Please feel free reopen the issue if there's any problem, thank you 😄
